crates/oxc_linter/src/rules/jest/prefer_to_be.rs

```diff
@@ -0,0 +1,511 @@
+    #[diagnostic(severity(warning))]
+    UseToBeNull(#[label] Span),
+    #[error("eslint-plugin-jest(prefer-to-be): Use `toBeNaN` instead.")]
+    #[diagnostic(severity(warning))]
+    UseToBeNaN(#[label] Span),
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct PreferToBe;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// When asserting against primitive literals such as numbers and strings, the
+    /// equality matchers all operate the same, but read slightly differently in code.
+    ///
+    /// This rule recommends using the `toBe` matcher in these situations, as it forms
+    /// the most grammatically natural sentence. For `null`, `undefined`, and `NaN` this
+    /// rule recommends using their specific `toBe` matchers, as they give better error
+    /// messages as well.
+    ///
+    /// ### Example
+    ///
+    /// ```javascript
+    /// // valid
+    /// expect(value).not.toBe(5);
+    /// expect(getMessage()).toBe('hello world');
+    /// expect(loadMessage()).resolves.toBe('hello world');
+    /// expect(didError).not.toBe(true);
+    /// expect(catchError()).toStrictEqual({ message: 'oh noes!' });
+    ///
+    /// // invalid
+    /// expect(value).not.toEqual(5);
+    /// expect(getMessage()).toStrictEqual('hello world');
+    /// expect(loadMessage()).resolves.toEqual('hello world');
+    /// ```
+    PreferToBe,
+    style,
+);
+
+impl Rule for PreferToBe {
+    fn run_once(&self, ctx: &LintContext) {
+        for possible_jest_node in &collect_possible_jest_call_node(ctx) {
+            Self::run(possible_jest_node, ctx);
+        }
+    }
+}
+
+impl PreferToBe {
+    fn run<'a>(possible_jest_node: &PossibleJestNode<'a, '_>, ctx: &LintContext<'a>) {
+        let node = possible_jest_node.node;
+        let AstKind::CallExpression(call_expr) = node.kind() else {
+            return;
+        };
+        let Some(jest_expect_fn_call) =
+            parse_expect_jest_fn_call(call_expr, possible_jest_node, ctx)
+        else {
+            return;
+        };
+        let Some(matcher) = jest_expect_fn_call.matcher() else {
+            return;
+        };
+
+        let has_not_modifier = jest_expect_fn_call
+            .modifiers()
+            .iter()
+            .filter(|modifier| modifier.is_name_equal("not"))
+            .count()
+            > 0;
+
+        if has_not_modifier {
+            if matcher.is_name_equal("toBeUndefined") {
+                Self::check_and_fix("Defined", call_expr, matcher, &jest_expect_fn_call, ctx);
+                return;
+            } else if matcher.is_name_equal("toBeDefined") {
+                Self::check_and_fix("Undefined", call_expr, matcher, &jest_expect_fn_call, ctx);
+                return;
+            }
+        }
+
+        if !is_equality_matcher(matcher) || jest_expect_fn_call.args.len() == 0 {
+            return;
+        }
+
+        let first_matcher_arg = get_first_matcher_arg(&jest_expect_fn_call);
```

Comment:
Seems like `get_inner_expression` is enough.

Reply:
Yes. Update.